src/features/governance/hooks/useProposalQuorum.tsx

```diff
@@ -15,19 +19,44 @@ interface ParticipationParameters {
 
 export function useProposalQuorum(propData?: MergedProposalData) {
   const participationParameters = useParticipationParameters();
+  const thresholds = useThresholds(propData?.proposal);
 
-  if (!propData?.proposal?.networkWeight) return undefined;
+  if (!propData || !propData.proposal) return;
+  if (!thresholds || !participationParameters) return;
 
-  return BigInt(
-    BigNumber(propData.proposal.networkWeight.toString())
-      .times(participationParameters.baseline)
-      .times(participationParameters.baselineQuorumFactor)
-      .toFixed(0),
+  // https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/governance/Governance.sol#L1567
+  let quorum = new BigNumber(participationParameters.baseline).times(
+    participationParameters.baselineQuorumFactor,
   );
+  // https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/governance/Proposals.sol#L195-L211
+  let requiredVotes = BigInt(quorum.times(propData.proposal.networkWeight.toString()).toFixed(0));
+
+  // It would seem calculating `support` is unecessary ?
+  // const totalVotes =
+  //   propData.proposal.votes[VoteType.Abstain] +
+  //   propData.proposal.votes[VoteType.No] +
+  //   propData.proposal.votes[VoteType.Yes];
+
+  // let noVotesWithPadding = propData.proposal.votes[VoteType.No];
+  // if (requiredVotes > totalVotes) {
+  //   noVotesWithPadding += requiredVotes - totalVotes;
+  // }
+
+  // const support = new BigNumber(propData.proposal.votes[VoteType.Yes].toString())
+  //   .div(
+  //     new BigNumber(propData.proposal.votes[VoteType.Yes].toString()).plus(
+  //       noVotesWithPadding.toString(),
+  //     ),
+  //   )
+  //   .toNumber();
+
+  const maxThreshold = Math.max(...thresholds);
```

Comment:
hmm shouldnt the threashold use be determined by what contract calls are to be made? not just the max

Reply:
we're getting all the contract calls from the proposals in `useThresholds` but eventually we only really care about the "hardest one to pass", right?

Reply:
ok you are correct. I was thinking this was the absolute max not the highest for the transactions being called